core/rawdb/prunedfreezer.go

```diff
@@ -0,0 +1,286 @@
+package rawdb
+
+import (
+	"math"
+	"os"
+	"path/filepath"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethdb"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/prometheus/tsdb/fileutil"
+)
+
+// nodatafreezer is an empty freezer, only record 'frozen' , the next recycle block number form kvstore.
+type prunedfreezer struct {
```

Comment:
What is the name of this freezer,  `prunedfreezer` or `nodatafreezer`? Prefer `emptyFreezer` personaly.